src/main/java/org/opentripplanner/model/StopTime.java

```diff
@@ -275,4 +276,37 @@ public String toString() {
                 + getTrip().getId() + " times=" + TimeUtils.timeToStrLong(getArrivalTime())
                 + "-" + TimeUtils.timeToStrLong(getDepartureTime()) + ")";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) { return true; }
+        if (!(o instanceof StopTime)) { return false; }
+        final StopTime stopTime = (StopTime) o;
+        return arrivalTime == stopTime.arrivalTime
+                && departureTime == stopTime.departureTime
+                && timepoint == stopTime.timepoint
+                && stopSequence == stopTime.stopSequence
+                && pickupType == stopTime.pickupType
+                && dropOffType == stopTime.dropOffType
+                && Double.compare(stopTime.shapeDistTraveled, shapeDistTraveled) == 0
+                && flexWindowStart == stopTime.flexWindowStart
+                && flexWindowEnd == stopTime.flexWindowEnd
+                && flexContinuousPickup == stopTime.flexContinuousPickup
+                && flexContinuousDropOff == stopTime.flexContinuousDropOff
+                && Objects.equals(stop, stopTime.stop)
+                && Objects.equals(stopHeadsign, stopTime.stopHeadsign)
+                && Objects.equals(routeShortName, stopTime.routeShortName)
+                && Objects.equals(farePeriodId, stopTime.farePeriodId)
+                && Objects.equals(bookingInfo, stopTime.bookingInfo);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                stop, arrivalTime, departureTime, timepoint, stopSequence, stopHeadsign,
+                routeShortName, pickupType, dropOffType, shapeDistTraveled, farePeriodId,
+                flexWindowStart, flexWindowEnd, flexContinuousPickup, flexContinuousDropOff,
+                bookingInfo
+        );
+    }
```

Comment:
Unsturtent if this is a good idea or not.

Reply:
The default for OTP has heavily leaned toward identity equality, so the presence of any equals or hashcode requires an explanation in comments. It may also run up against assumptions elsewhere in the system - we can discuss this today on the call.